ShoeBox/Classes/Helpers/LocationManager/LocationManager.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  LocationManager.swift
+//  ShoeBox
+//
+//  Created by Madalina Miron on 28/11/16.
+//  Copyright © 2016 ShoeBox. All rights reserved.
+//
+
+import Foundation
+import CoreLocation
+
+protocol LocationManagerDelegate {
+    func locationReceived(currentLocation: CLLocation)
+    func locationDidFailWithError(error: NSError)
+}
+
+class LocationManager: NSObject, CLLocationManagerDelegate {
+    static let sharedInstance = LocationManager()
```

Comment:
Not really sure how I would do that. I think static is already lazy. Can you give an example ?